sqlglot/parser.py

```diff
@@ -923,6 +923,9 @@ class Parser(metaclass=_Parser):
         TokenType.START_WITH: lambda self: ("connect", self._parse_connect()),
     }
 
+    # For concrete implementation, look at tsql
+    OPTION_PREFIX_TO_PARSE_FUNCTION: t.Dict[str, t.Callable] = {}
```

Comment:
Hey @georgesittas, I think there's something we need to decide here in regards to whether OPTION is a generic SQL feature or not. In practice, I've only seen it in TSQL (Oracle has a similar feature, with specially-crafted comments that define similar behaviour). As a TSQL-only feature, I wanted to implement this in the TSQL parser and moved it up to the generic one as requested.
Can you please share your thoughts on whether this feature belongs in the TSQL or generic parser?

Reply:
Even though we occasionally implement new features in the base classes to make them more "generic", imo supporting `OPTION` in particular only for T-SQL seemed more reasonable because the options are T-SQL-specific. @tobymao may have a different opinion here, so I'll defer to him.

My intention from the start was to try and keep this in T-SQL, apologies if there was a misunderstanding w.r.t.

> I wanted to implement this in the TSQL parser and moved it up to the generic one as requested.

Let's connect in [Slack](https://tobikodata.com/slack) to iterate / chat more easily?